pilot/pkg/bootstrap/server.go

```diff
@@ -278,18 +278,28 @@ func NewServer(args *PilotArgs) (*Server, error) {
 		return nil, fmt.Errorf("error initializing secure gRPC Listener: %v", err)
 	}
 
+	var wh *inject.Webhook
 	// common https server for webhooks (e.g. injection, validation)
-	s.initSecureWebhookServer(args)
-
-	wh, err := s.initSidecarInjector(args)
-	if err != nil {
-		return nil, fmt.Errorf("error initializing sidecar injector: %v", err)
+	if s.kubeClient != nil || args.isTestingServer() {
```

Comment:
Is it possible to mock a client instead of hacking option?

Reply:
Removed the hack and passed as an initializer func.